resources/js/script.js

```diff
@@ -26,145 +28,173 @@ $(".add-to-cart").click(function(event){
-	var cartArray = listCart();
+	var cartArray = shoppingCart.listCart();
 	var output = "";
+	var position = 1;
 	for (var i in cartArray) {
-		output += "<li>"+cartArray[i].name+" "+"<span>"+cartArray[i].count+"</span>"+"</li>";
+		output += "<tr>"+"<td>"+"<button class='delete-item' data-name='"+cartArray[i].name+"'>X</button>"+"<p>"+position+"</p>"+"</td>"+"<td>"+cartArray[i].name+"</td>"+"<td>"+"<span>"+cartArray[i].count+"</span>"+"<div class='add-remove-buttons'>"+" <button class='plus-item' data-name='"+cartArray[i].name+"'> + </button> "+"<button class='substract-item' data-name='"+cartArray[i].name+"'> - </button> "+"</div>"+"</td>"+"<td>"+cartArray[i].price+"</td>"+"<td>"+cartArray[i].total+"</td>"+"</tr>";
+		
+		position ++;
 	}
 	$("#show-cart").html(output);
-	$("#total-cart").html(totalCart());
-};
+	$("#count-cart").html(shoppingCart.countCart())
+	$("#total-cart").html(shoppingCart.totalCart());
+}
+
+$("#show-cart").on("click", ".delete-item", function(event) {
+	var name = $(this).attr("data-name");
+	shoppingCart.removeItemFromCartAll(name);
+	displayCart();
+});
+
+$("#show-cart").on("click", ".plus-item", function(event) {
+	var name = $(this).attr("data-name");
+	shoppingCart.addItemToCart(name, 0, 1);
+	displayCart();
+});
+
+$("#show-cart").on("click", ".substract-item", function(event) {
+	var name = $(this).attr("data-name");
+	shoppingCart.removeItemFromCart(name);
+	displayCart();
+});
+
+
 
 /***************************************/
 // Shopping cart functions
 /* Scope - Determins where a variable or a value is visible */ 
 
-var cart = [];
-//when you use the sintax below, it actually generates an object
+var shoppingCart = {};
+shoppingCart.cart = [];
 
-var Item = function(name, price, count) {
+shoppingCart.Item = function(name, price, count) {
 	this.name = name
 	this.price = price
 	this.count = count
 };
 
-/*
-// new item for shopping cart
-var apple = new Item("Apple", 1.99, 1); // {name, price, count}
-
-cart.push(new Item("Brush", 2.13, 1));
-cart.push(apple);*/
-
-// addItemToCart(name, price, count)
-
-function addItemToCart(name, price, count) {
-	for (var i in cart) {
-		if (cart[i].name === name) {
-			cart[i].count += count;
-			saveCart();
+shoppingCart.addItemToCart = function(name, price, count) {
+	for (var i in this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart[i].count += count;
+			this.saveCart();
 			return;
 		}
 	}
-	var item = new Item(name, price, count);
-	cart.push(item);
-	saveCart();
+	var item = new this.Item(name, price, count);
+	this.cart.push(item);
+	this.saveCart();
 };
 
-// removeItemFromCart(name) // Removes one item
-
-function removeItemFromCart(name) {
-	for (var i in  cart) {
-		if (cart[i].name === name) {
-			cart[i].count --;
-			if (cart[i].count === 0) {
-				cart.splice(i, 1); //it removes "1" item from the array
+shoppingCart.removeItemFromCart = function(name) {
+	for (var i in  this.cart) {
+		if (this.cart[i].name === name) {
```

Comment:
why not `==`?

Reply:
Well, i don't really have a reason, I use "===" because it's more strict and it also compares the type so when i have to check if 2 variables are equal I made a habit of using "===". Although it's true that in this situation it would work perfectly fine with "==" too.

Reply:
Haha @AdrianFlorian, nice. Great @AlexHoisan ! Keep using the strict version 👏 🏅 Using the loose version can result in some nasty bugs.